airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/JobCreationAndStatusUpdateActivityImpl.java

```diff
@@ -333,6 +341,55 @@ public void ensureCleanJobState(final EnsureCleanJobStateInput input) {
     failNonTerminalJobs(input.getConnectionId());
   }
 
+  @Override
+  public boolean isLastJobOrAttemptFailure(JobCheckFailureInput input) {
+    final int limit = 2;
+    boolean lastAttemptCheck = false;
+    boolean lastJobCheck = false;
+
+    Set<JobConfig.ConfigType> configTypes = new HashSet<>();
+    configTypes.add(SYNC);
+
+    try {
+      List<Job> jobList = jobPersistence.listJobsIncludingId(configTypes, input.getConnectionId().toString(), input.getJobId(), limit);
```

Comment:
To confirm: limit will query SQL for  2, but are we ordering by the right thing to get the most recent 2 attempts?

Reply:
Good question. Judging by the way the structure of the method for obtaining jobs is built, the first request will be to get OffsetDateTime of creation of an active job. With the second request, we will receive the number of jobs whose creation date is greater than or equal to the result of the first request. Since we make a request with the identifier of the last active job, the number we will receive from the second request will always be 1. And then, during the calculation, we will receive the number of jobs that will be returned to us for processing - this will be calculated based on the result of the second request, and the limit we passed to this request - the result will always be 2.

![2022-10-17_11-03](https://user-images.githubusercontent.com/88329385/196136410-03a373a1-fea9-4970-b0dd-0e930df74d64.png)